src/OpenTelemetry.Instrumentation.Runtime/description.md

```diff
@@ -0,0 +1,175 @@
+
+- [GC.CollectionCount](https://docs.microsoft.com/dotnet/api/system.gc.collectioncount):
+The number of times garbage collection has occurred for the specified generation
+of objects.
+
+### Additional GC metrics only available for NETCOREAPP3_1_OR_GREATER
+
+| Name                                             | Description                                      | Units | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values           |
+|--------------------------------------------------|--------------------------------------------------|-------|-------------------|------------|------------------|----------------------------|
+| process.runtime.dotnet.**gc.allocated.bytes**    | Bytes allocated over the lifetime of the process | `By`  | ObservableCounter | `Int64`    |                  |                            |
+| process.runtime.dotnet.**gc.fragmentation.size** | GC fragmentation size                            | `By`  | ObservableGauge   | `Int64`    | gen              | gen0, gen1, gen2, loh, poh |
+
+- [GC.GetTotalAllocatedBytes](https://docs.microsoft.com/dotnet/api/system.gc.gettotalallocatedbytes):
+  Gets a count of the bytes allocated over the lifetime of the process. The returned
+value does not include any native allocations. The value is an approximate count.
+
+  - [GCGenerationInfo.FragmentationAfterBytes Property](https://docs.microsoft.com/dotnet/api/system.gcgenerationinfo.fragmentationafterbytes#system-gcgenerationinfo-fragmentationafterbytes)
+  Gets the fragmentation in bytes on exit from the reported collection.
+
+### Additional GC metrics only available for NET6_0_OR_GREATER
+
+| Name                                    | Description        | Units | Instrument Type | Value Type | Attribute Key(s) | Attribute Values           |
+|-----------------------------------------|--------------------|-------|-----------------|------------|------------------|----------------------------|
+| process.runtime.dotnet.**gc.committed** | GC Committed Bytes | `By`  | ObservableGauge | `Int64`    |                  |                            |
+| process.runtime.dotnet.**gc.heapsize**  |                    | `By`  | ObservableGauge | `Int64`    | gen              | gen0, gen1, gen2, loh, poh |
+
+- [GCMemoryInfo.TotalCommittedBytes](https://docs.microsoft.com/dotnet/api/system.gcmemoryinfo.totalcommittedbytes?view=net-6.0#system-gcmemoryinfo-totalcommittedbytes):
+Gets the total committed bytes of the managed heap.
+
+- [GC.GetGCMemoryInfo().GenerationInfo[i].SizeAfterBytes](https://docs.microsoft.com/dotnet/api/system.gcgenerationinfo):
+Represents the size in bytes of a generation on exit of the GC reported in GCMemoryInfo.
+
+## JIT Compiler related metrics
+
+The metrics in this section can be enabled by setting the
+`RuntimeMetricsOptions.IsJitEnabled` switch.
+
+These metrics are only available for NET6_0_OR_GREATER.
```

Comment:
Using the preprocessor directives in this doc like `NET6_0_OR_GREATER` is very technical. Maybe just say like...

`These metrics are available when targeting .NET6 and greater`.

Reply:
I think this is opened to just solicit feedback only.